frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditLayoutSetForSubform/EditLayoutSet/CreateNewSubformLayoutSet/SubformDataModelSelect.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import {
+  convertMetadataListToOptions,
+  findMetadataOptionByRelativeUrl,
+} from 'app-development/utils/metadataUtils';
+import type { MetadataOption } from 'app-development/types/MetadataOption';
+import classes from './SubformDataModelSelect.module.css';
+import type { DataModelMetadataJson } from 'app-shared/types/DataModelMetadata';
+import { useTranslation } from 'react-i18next';
+import { StudioNativeSelect } from '@studio/components';
+
+export interface ISubformDataModelSelectProps {
+  dataModels: DataModelMetadataJson[];
+  disabled: boolean;
+  selectedOption: MetadataOption | null;
+  setSelectedOption: (option: MetadataOption) => void;
+}
+
+export const SubformDataModelSelect = ({
+  dataModels,
+  disabled,
+  selectedOption,
+  setSelectedOption,
+}: ISubformDataModelSelectProps) => {
+  const { t } = useTranslation();
+  const options = convertMetadataListToOptions(dataModels);
+
+  function handleChange(e: React.ChangeEvent<HTMLSelectElement>) {
```

Comment:
We don't need the whole event here, so I would suggest passing only the value:

```suggestion
  function handleChange(selectedDataModel: string) {
```

Reply:
Fixed